src/snowcli/app/commands_registration/command_plugins_loader.py

```diff
@@ -0,0 +1,113 @@
+import logging
+from typing import List, Optional, Dict
+
+import pluggy
+
+from snowcli.api.plugin.command import (
+    SNOWCLI_COMMAND_PLUGIN_NAMESPACE,
+    plugin_hook_specs,
+    CommandSpec,
+    CommandPath,
+)
+from snowcli.app.commands_registration import (
+    LoadedCommandPlugin,
+    LoadedBuiltInCommandPlugin,
+)
+from snowcli.app.commands_registration.builtin_plugins import (
+    builtin_plugin_name_to_plugin_spec,
+)
+from snowcli.cli.exception_logging import exception_logging
+
+log = logging.getLogger(__name__)
+log_exception = exception_logging(log)
+
+
+class CommandPluginsLoader:
+    def __init__(self):
+        plugin_manager = pluggy.PluginManager(SNOWCLI_COMMAND_PLUGIN_NAMESPACE)
+        plugin_manager.add_hookspecs(plugin_hook_specs)
+        self._plugin_manager = plugin_manager
+        self._loaded_plugins: Dict[str, LoadedCommandPlugin] = {}
+        self._loaded_command_paths: Dict[CommandPath, LoadedCommandPlugin] = {}
+
+    def register_only_builtin_plugins(self) -> None:
+        for (plugin_name, plugin) in builtin_plugin_name_to_plugin_spec.items():
+            try:
+                self._plugin_manager.register(plugin=plugin, name=plugin_name)
+            except Exception as ex:
+                log_exception(
+                    f"Cannot register plugin [{plugin_name}]: {ex.__str__()}", ex
+                )
+
+    def load_all_registered_plugins(self) -> List[LoadedCommandPlugin]:
+        for (plugin_name, plugin) in self._plugin_manager.list_name_plugin():
+            self._load_plugin(plugin_name, plugin)
+        return list(self._loaded_plugins.values())
+
+    def _load_plugin(self, plugin_name: str, plugin) -> Optional[LoadedCommandPlugin]:
+        already_loaded_plugin = self._loaded_plugins.get(plugin_name)
+        if already_loaded_plugin:
+            return already_loaded_plugin
+        else:
+            return self._load_new_plugin(plugin_name, plugin)
```

Comment:
```suggestion
        return self._load_new_plugin(plugin_name, plugin)
```

Reply:
I'm not sure if your version is easier to read but I applied your suggestion ;)